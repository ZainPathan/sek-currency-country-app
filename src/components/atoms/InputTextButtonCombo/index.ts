import InputTextButtonCombo from './InputTextButtonCombo';

export {
    InputTextButtonCombo
};

export default InputTextButtonCombo;