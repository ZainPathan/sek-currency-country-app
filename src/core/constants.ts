export const labels = {
    header: 'Swedish Krona (SEK) Currency Converter to Local Currency',
    searchCountryPlaceholderText: 'Search Country',
    searchCountryButtonText: 'Search Country',
    amountInSEKPlaceholder: 'Currency Amount in SEK',
    getAmountInCurrenciesButtonText: 'Get Amount in Local Currencies',
    countryDetailsHeaders: [
        'Country Flag',
        'Country Name',
        'Capital',
        'Population',
        'Currency',
        'Amount in Local Currency'
    ],
    searchCountryError: 'Unable to search country',
};

export const config = {
    fixerCurrencyAPIKey: '288b1f30e9e9881212f483d3768d17aa',
    baseCurrencyCode: 'EUR'
}