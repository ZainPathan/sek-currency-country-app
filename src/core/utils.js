export const GetAmountInCurrency = (currencyMap, currencyCode, currencySymbol, amountInSEK) => {
    return ' ' + currencySymbol + ' ' + (currencyMap[currencyCode] * amountInSEK).toFixed(2);
};