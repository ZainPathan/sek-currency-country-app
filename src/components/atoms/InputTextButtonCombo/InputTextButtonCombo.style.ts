import { makeStyles } from '@material-ui/core/styles';

const useInputTextButtonComboStyles = makeStyles({
    inputField: {
        height: 20,
        width: '100%',
    },
    button: {
        height: 60,
        width: '100%',
        marginTop: 60
    }
});

export default useInputTextButtonComboStyles;