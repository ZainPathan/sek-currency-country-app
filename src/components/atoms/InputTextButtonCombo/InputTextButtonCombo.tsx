import React from 'react';
import { Button, TextField } from '@material-ui/core';
import useInputTextButtonComboStyles from './InputTextButtonCombo.style';

const InputTextButtonCombo = (props) => {
    const classes = useInputTextButtonComboStyles();

    return (
        <>
            <TextField
                id="outlined-basic"
                label={props.inputPlaceholder}
                variant="outlined"
                className={classes.inputField}
                onChange={props.onChange}
            />
            <Button
                variant="contained"
                color="primary"
                className={classes.button}
                onClick={props.onClick}
                disabled={props.disabled}
            >
                {props.buttonLabel}
            </Button>
        </>
    );
};

export default InputTextButtonCombo