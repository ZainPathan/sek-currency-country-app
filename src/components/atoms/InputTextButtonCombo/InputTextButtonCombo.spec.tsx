import React from 'react';
import renderer from 'react-test-renderer';
import InputTextButtonCombo from './InputTextButtonCombo';
import { labels } from '../../../core/constants';
import { fireEvent, render } from '@testing-library/react';

describe('InputTextButtonCombo tests', () => {

    const inputPlaceholder = labels.searchCountryPlaceholderText;
    const buttonLabel = labels.searchCountryButtonText;
    const onChange = jest.fn();
    const onClick = jest.fn();

    it('renders properly', () => {
        const tree = renderer
            .create(
                <InputTextButtonCombo
                    inputPlaceholder={inputPlaceholder}
                    buttonLabel={buttonLabel}
                    onChange={onChange}
                    onClick={onClick}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with input text placeholder', () => {
        const inputPlaceholderText = 'Test Input Placeholder';
        const { getAllByText } = render(
            <InputTextButtonCombo
                inputPlaceholder={inputPlaceholderText}
                buttonLabel={buttonLabel}
                onChange={onChange}
                onClick={onClick}
            />
        );
        expect(getAllByText(inputPlaceholderText)).toBeTruthy();
    });

    it('renders and button click event handler is called', () => {
        const { getByRole } = render(
            <InputTextButtonCombo
                inputPlaceholder={inputPlaceholder}
                buttonLabel={buttonLabel}
                onChange={onChange}
                onClick={onClick}
            />
        );
        fireEvent.click(getByRole('button'));
        expect(onClick).toHaveBeenCalled();
    });
});