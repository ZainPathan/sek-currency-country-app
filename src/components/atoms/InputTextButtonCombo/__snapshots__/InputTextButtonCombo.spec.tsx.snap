// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextButtonCombo tests renders properly 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-inputField-1"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-basic"
      id="outlined-basic-label"
    >
      Search Country
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-5"
        >
          <span>
            Search Country
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    />
  </button>,
]
`;
